### PR TITLE
NEO-1845: show menu above toggle when there are no space below and enough space above

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[Link to updated components in Deploy Preview](https://deploy-preview-266--neo-react-library-storybook.netlify.app/?path=/story/components-menu--up-going-menus)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

This the first part of the ticket, i.e. Place the menu on the top of the toggle if possible.

### CODE REVIEW

**Review by joe-s-avaya — APPROVED** [review]
Nice! Lots of tests and stories written 💯  and the implementation code all looks solid 🏅

File: src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -83,34 +82,70 @@ export const Menu = forwardRef(
...
+    const toggleRef = ref || _ref;
+    const menuRef = useRef<HTMLDivElement>(null);
 
     const [isOpen, setOpen] = useState(defaultIsOpen);
     const [enterCounter, setEnterCounter] = useState(1);
-    const clonedChildren = useMemo(
-      () => addIdToChildren(children, SubMenu.name),
-      [children],
-    );
+    const [up, setUp] = useState(false);
```
Comment (enrique-prado) [inline src/components/Menu/Menu.tsx:91]: Nit: up / setUp names seem a bit ambiguous (especially setUp).  How about renderUp / setRenderUp... or upwards / setUpwards.?

**Review by enrique-prado — APPROVED** [review]
Just the one nit.  Very nice work!

File: src/components/Menu/Menu.tsx (hunk 1 above)
Comment (yangAtSpoken) [reply src/components/Menu/Menu.tsx]: changed, thx.

**Comment by matthewAvaya** [comment]
- Clicking on the **UP** arrow on the keyboard opens the dropdown menu at the bottom of the trigger rather than opening the last item on the list. 
Screen Shot 2023-09-29 at 10 14 45 AM

- In the same scenario above using the UP arrow, when using the screen reader, it doesn't read any of the individual menu item as none of the individual ite
Screen Shot 2023-09-29 at 11 13 58 AM
m is selected. Instead the entire dropdown is focused. 


- When clicking on the ESC key on the keyboard to close the dropdown menu, there's a black outline box from the focus state that is still visible. I'm not sure if that is controllable by us or not but the expectation is to remove that focus (black box). 
Screen Shot 2023-09-29 at 10 39 31 AM

- When using screen reader,

**Comment by yangAtSpoken** [comment]
> * Clicking on the **UP** arrow on the keyboard opens the dropdown menu at the bottom of the trigger rather than opening the last item on the list.
> 
> Screen Shot 2023-09-29 at 10 14 45 AM
> * In the same scenario above using the UP arrow, when using the screen reader, it doesn't read any of the individual menu item as none of the individual ite
> 
> Screen Shot 2023-09-29 at 11 13 58 AM m is selected. Instead the entire dropdown is focused.
> * When clicking on the ESC key on the keyboard to close the dropdown menu, there's a black outline box from the focus state that is still visible. I'm not sure if that is controllable by us or not but the expectation is to remove that focus (black box).
> 
> Screen Shot 2023-09-29 at 10 39 31 AM
> * When using screen reader,

@matthewAvaya 1st issue fixed. 2nd one: is VoiceOver navigation issue. Looks like you went into/out of a menu group.  When VoiceOver works for me, I could just use enter and arrows navigate through the multilevel menu.  3rd issue is a VoiceOver issue. The box means VoiceOver is waiting for the next element to focus on.

**Review by matthewAvaya — APPROVED** [review]
Looks great!

**Pull request merged**